core/Azure.Mcp.Core/src/Commands/CommandFactory.cs

```diff
@@ -130,23 +127,27 @@ private void RegisterCommandGroup()
                 throw error;
             }
 
-            // Create a temporary root node to register all the area's subgroups to.
-            // Use the root to map commands back to their area.
-            var tempRoot = new CommandGroup(RootCommandGroupName, string.Empty);
+            // Get the commands for the IAreaSetup and register it to the root node.
+            var commandTree = area.RegisterCommands(_serviceProvider);
+            _rootGroup.AddSubGroup(commandTree);
```

Comment:
To me, it is interesting to see vars with _ prefix and some without.  Maybe clean up any file for this PR?  If not ok, to file issue for later